browsers/duckduckgo.safariextension/js/companies.js

```diff
@@ -0,0 +1,156 @@
+                lastStatsResetDate: lastStatsResetDate
+            }
+        },
+
+        setTotalPagesFromStorage: (n) => {
+            if (n) totalPages = n
+        },
+
+        setTotalPagesWithTrackersFromStorage: (n) => {
+            if (n) totalPagesWithTrackers = n
+        },
+
+        resetData: () => {
+            companyContainer = {}
+            topBlocked.clear()
+            totalPages = 0
+            totalPagesWithTrackers = 0
+            lastStatsResetDate = Date.now()
+            Companies.syncToStorage()
+        },
+
+        getLastResetDate: ()  => lastStatsResetDate,
+
+        incrementTotalPages: () => {
+            totalPages += 1
+            Companies.syncToStorage()
+        },
+
+        incrementTotalPagesWithTrackers: () => {
+            totalPagesWithTrackers += 1
+            Companies.syncToStorage()
+        },
+
+        syncToStorage: () => {
+            var toSync = {};
+            toSync[storageName] = companyContainer;
+            utils.syncToStorage(toSync)
+            utils.syncToStorage({'totalPages': totalPages})
+            utils.syncToStorage({'totalPagesWithTrackers': totalPagesWithTrackers})
+            utils.syncToStorage({'lastStatsResetDate': lastStatsResetDate})
+        },
+
+        buildFromStorage: () => {
+            utils.getFromStorage(storageName, function (storageData) {
+                for (company in storageData) {
+                    let newCompany = Companies.add(company)
+                    newCompany.set('count', storageData[company].count || 0)
+                    newCompany.set('pagesSeenOn', storageData[company].pagesSeenOn || 0)
+                }
+            })
+
+            utils.getFromStorage('totalPages', (n) => { if (n) totalPages = n })
+            utils.getFromStorage('totalPagesWithTrackers', (n) => { if (n) totalPagesWithTrackers = n })
+            utils.getFromStorage('lastStatsResetDate', (d) => {
+                if (d) {
+                    lastStatsResetDate = d
+                } else {
+                    // if 'lastStatsResetDate' not found, reset all data
+                    // https://app.asana.com/0/0/460622849089890/f
+                    Companies.resetData()
+                }
+            })
+         }
+     }
+})()
+
+Companies.buildFromStorage()
+
+/*
+// sync data to storage when a tab finishes loading
+chrome.tabs.onUpdated.addListener( (id,info) => {
+    if (info.status === "complete") {
+        Companies.syncToStorage()
+    }
+})
+
+chrome.runtime.onMessage.addListener((req, sender, res) => {
+    if (req.getTopBlocked) {
+        res(Companies.getTopBlocked(req.getTopBlocked))
+    } else if (req.getTopBlockedByPages) {
+        res(Companies.getTopBlockedByPages(req.getTopBlockedByPages))
+    } else if (req.resetTrackersData) {
+        Companies.resetData()
+        chrome.runtime.sendMessage({'didResetTrackersData': Companies.getLastResetDate()})
+        res()
+    }
+    return true
+})
+*/
```

Comment:
The only difference here is the chrome listeners are commented out. Is that logic covered somewhere else? And if that's going to be the only difference, can we remove this file and add a guard to the other one, e.g. `if (window.chrome) { do chrome.* stuff }`